### PR TITLE
feat(replays): switch to replay_id column from tag

### PR DESCRIPTION
**Pull request by JoshFerge**
We want to have the replay_id column be queried on instead of the tag that we previously had, as we no longer are populating the tag for non javascript platforms. Depends on https://github.com/getsentry/snuba/pull/5777.

### CODE REVIEW

File: src/sentry/search/events/builder/discover.py [hunk 1]
```diff
@@ -1578,14 +1575,16 @@ def default_filter_converter(
...
+        if name in {"id", "trace", "profile.id", "replay.id"}:
             if search_filter.value.is_wildcard():
                 raise InvalidSearchQuery(WILDCARD_NOT_ALLOWED.format(name))
             elif not search_filter.value.is_event_id():
                 if name == "trace":
                     label = "Filter Trace ID"
                 elif name == "profile.id":
                     label = "Filter Profile ID"
+                # elif name == "replay.id":
+                #     label = "Filter Replay ID"
```
Comment (cmanallen) [inline src/sentry/search/events/builder/discover.py:1587]: Does this need to be uncommented?

File: tests/sentry/replays/test_organization_replay_count.py [hunk 2]
```diff
@@ -413,36 +416,6 @@ def test_simple_transaction(self):
         assert response.status_code == 200, response.content
         assert response.data == expected
 
-    def test_max_51(self):
```
Comment (cmanallen) [inline tests/sentry/replays/test_organization_replay_count.py:416]: Is this coverage no longer applicable?

File: src/sentry/snuba/events.py [hunk 3]
```diff
@@ -742,8 +742,20 @@ class Columns(Enum):
...
+        transaction_name="replay_id",
+        discover_name="replay_id",
+        issue_platform_name="replay_id",
+        alias="replay.id",
+    )
+    # We used to set the replay_id as a tag on error events as
+    # replayId. We allow this query for backwards compatibility,
+    # but in the future shouldn't be displayed in the UI anywhere
+    # as a suggested column.
+    REPLAY_ID_DEPRECATED = Column(
```
Comment (cmanallen) [inline src/sentry/snuba/events.py:754]: Where is this used?

**Review by cmanallen — APPROVED** [review]

File: src/sentry/replays/usecases/replay_counts.py [hunk 4]
```diff
@@ -9,8 +9,11 @@
 from sentry.api.event_search import ParenExpression, SearchFilter, parse_search_query
 from sentry.models.group import Group
 from sentry.replays.query import query_replays_count
-from sentry.search.events.builder import QueryBuilder
-from sentry.search.events.types import QueryBuilderConfig, SnubaParams
+from sentry.search.events.types import SnubaParams
+from sentry.snuba import discover, issue_platform
+
+# from sentry.search.events.builder import QueryBuilder
+# from sentry.search.events.types import QueryBuilderConfig, SnubaParams
```
Comment (wedamija) [inline src/sentry/replays/usecases/replay_counts.py:16]: Nit: remove

File: src/sentry/replays/usecases/replay_counts.py [hunk 5]
```diff
@@ -62,6 +66,12 @@ def _get_replay_id_mappings(
     If select_column is replay_id, return an identity map of replay_id -> [replay_id].
     The keys of the returned dict are UUIDs, represented as 32 char hex strings (all '-'s stripped)
     """
+
+    if data_source == Dataset.Discover:
+        search_query_func = discover.query
+    elif data_source == Dataset.IssuePlatform:
+        search_query_func = issue_platform.query  # type: ignore[assignment]
```
Comment (wedamija) [inline src/sentry/replays/usecases/replay_counts.py:73]: Just checking we actually write to this col from issue platform?

**Review by wedamija — COMMENTED** [review]
Are you doing any sort of backfill here? Or will querying on this col this return no results for older rows?

File: tests/sentry/replays/test_organization_replay_count.py (hunk 2 above)
Comment (JoshFerge) [reply tests/sentry/replays/test_organization_replay_count.py:416]: woops, meant to comment here. its a really slow test, and i don't think gets us much.

**Comment by JoshFerge** [comment]
> Are you doing any sort of backfill here? Or will querying on this col this return no results for older rows?

we've been writing this column for a little over a year now, just haven't been querying on it since we had the tag as well.

File: src/sentry/search/events/builder/discover.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/search/events/builder/discover.py:1587]: yes, thank you.

File: src/sentry/replays/usecases/replay_counts.py (hunk 5 above)
Comment (JoshFerge) [reply src/sentry/replays/usecases/replay_counts.py:73]: yup https://github.com/getsentry/snuba/blob/8d4ea6917d5efef88fafe5ca33814dbf91d02595/snuba/datasets/processors/search_issues_processor.py#L203

**Comment by wedamija** [comment]
> > Are you doing any sort of backfill here? Or will querying on this col this return no results for older rows?
> 
> we've been writing this column for a little over a year now, just haven't been querying on it since we had the tag as well.

I see, I missed that the snuba pr is just adding the col to the merge table 👍

File: src/sentry/replays/usecases/replay_counts.py (hunk 5 above)
Comment (wedamija) [reply src/sentry/replays/usecases/replay_counts.py:70]: Ok - so as long as it's in the context for the event we pass to snuba it should work...

Are replays associated to transactions? If so, then perf issues from transactions should work automatically. Other issue platform types that make a "synthetic" event would need to explicitly include it there

Comment (JoshFerge) [reply src/sentry/replays/usecases/replay_counts.py:70]: yup, replays are associated to transactions. its done via the dynamic sampling context, which then adds the replay context. and for others:

rage click issue: https://github.com/getsentry/sentry/blob/8666d0c9f60be94988545522e1ff74aa7c0002f5/src/sentry/replays/usecases/ingest/issue_creation.py#L90

feedback issue:
https://github.com/getsentry/sentry/blob/8666d0c9f60be94988545522e1ff74aa7c0002f5/src/sentry/feedback/usecases/create_feedback.py#L98

Crons
-- not applicable

that should be everything

**Pull request merged**